File: 2026/Company/Company.Web/Modules/Sales/CategoriesType/CategoriesTypeRow.cs
using Company.Administration.Entities;

namespace Company.Sales;

[ConnectionKey("Sales"), Module("Sales"), TableName("[sales].[CategoriesType]")]
[DisplayName("Categories Type"), InstanceName("Categories Type")]
[ReadPermission(SalesPermissionKeys.Categories.View)]
[ModifyPermission(SalesPermissionKeys.Categories.Modify)]
[ServiceLookupPermission(SalesPermissionKeys.Categories.Delete)]
[LookupScript]
[LocalizationRow(typeof(CategoriesTypeLangRow), LocalizeListByDefault = true)]
public sealed class CategoriesTypeRow : LoggingRow<CategoriesTypeRow.RowFields>, IIdRow, INameRow, IIsActiveRow, IMultiTenantRow
{
    [DisplayName("Category Type Id"), Column("CategoryTypeID"), Identity, IdProperty]
    public int? CategoryTypeId { get => fields.CategoryTypeId[this]; set => fields.CategoryTypeId[this] = value; }

    [DisplayName("Type"), NotNull]
    public short? Type { get => fields.Type[this]; set => fields.Type[this] = value; }

    [DisplayName("Category Type"), Size(30), NotNull, QuickSearch, NameProperty]
    public string CategoryType { get => fields.CategoryType[this]; set => fields.CategoryType[this] = value; }


    #region Tenant & Activ

    [Insertable(false), Updatable(false)]
    public Int32? TenantId { get => fields.TenantId[this]; set => fields.TenantId[this] = value; }

    public Int32Field TenantIdField { get => fields.TenantId; }

    [NotNull, Insertable(false), Updatable(true)]
    public Int16? IsActive { get => fields.IsActive[this]; set => fields.IsActive[this] = value; }

    Int16Field IIsActiveRow.IsActiveField { get => fields.IsActive; }

    #endregion Tenant & Activ


    public class RowFields : LoggingRowFields
    {
        public Int32Field CategoryTypeId;
        public Int16Field Type;
        public StringField CategoryType;
        public Int16Field IsActive;
        public Int32Field TenantId;

    }
}

File: 2026/Company/Company.Web/Modules/ServerTypes/Sales/CategoriesTypeRow.ts
import { fieldsProxy, getLookup, getLookupAsync } from "@serenity-is/corelib";

export interface CategoriesTypeRow {
    CategoryTypeId?: number;
    Type?: number;
    CategoryType?: string;
    IsActive?: number;
    TenantId?: number;
    InsertUserId?: number;
    InsertDate?: string;
    UpdateUserId?: number;
    UpdateDate?: string;
}

export abstract class CategoriesTypeRow {
    static readonly idProperty = 'CategoryTypeId';
    static readonly isActiveProperty = 'IsActive';
    static readonly nameProperty = 'CategoryType';
    static readonly localTextPrefix = 'Sales.CategoriesType';
    static readonly lookupKey = 'Sales.CategoriesType';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<CategoriesTypeRow>('Sales.CategoriesType') }
    static async getLookupAsync() { return getLookupAsync<CategoriesTypeRow>('Sales.CategoriesType') }

    static readonly deletePermission = 'Company:Categories:Modify';
    static readonly insertPermission = 'Company:Categories:Modify';
    static readonly readPermission = 'Company:Categories:View';
    static readonly updatePermission = 'Company:Categories:Modify';

    static readonly Fields = fieldsProxy<CategoriesTypeRow>();
}